### ISSUE TITLE
ESM5/FESM5 imports are incorrect after updating to 3.0.0

### ISSUE THREAD

**Issue by maxtacco** [dherges/ng-packagr#915]
## Type of Issue

```
[ X] Bug Report
[ ] Feature Request
```

## Description

When building Angular library using `ng build` some of the resulting imports in ESM5/FESM5 bundles are incorrect after updating from 3.0.0-rc.5 to 3.0.0. ESM2015/FESM2015 imports are fine.

#### How To Reproduce

For example, I have the following generated using version 3.0.0:
```
...
import { ApiServices, ApiServiceSupport } from '@triu/common/api';
...
import { ApiServices as ApiServices$1 } from '@triu/common/api/src/api-services';
...

var CoreAccountService = /** @class */ (function (_super) {
    __extends(CoreAccountService, _super);
    ...
    CoreAccountService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    CoreAccountService.ctorParameters = function () { return [
        { type: ApiServices },
        { type: LocalStorageService }
    ]; };
    CoreAccountService.ngInjectableDef = defineInjectable({ factory: function CoreAccountService_Factory() { return new CoreAccountService(inject(ApiServices$1), inject(LocalStorageService)); }, token: CoreAccountService, providedIn: "root" });
    return CoreAccountService;
}(ApiServiceSupport));
```

And the following generated using version 3.0.0-rc.5:
```
...
import { ApiServices, ApiServiceSupport } from '@triu/common/api';
...
var CoreAccountService = /** @class */ (function (_super) {
    __extends(CoreAccountService, _super);
    ...
    CoreAccountService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    CoreAccountService.ctorParameters = function () { return [
        { type: ApiServices },
        { type: LocalStorageService }
    ]; };
    CoreAccountService.ngInjectableDef = defineInjectable({ factory: function CoreAccountService_Factory() { return new CoreAccountService(inject(ApiServices), inject(LocalStorageService)); }, token: CoreAccountService, providedIn: "root" });
    return CoreAccountService;
}(ApiServiceSupport));

```

Notice the second import in 3.0.0 version and how it is used in the generated service.

Also, the issues arises when I build an Angular app that references the generated libraries in tsconfig.json's 'paths' attribute:
```
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowUnreachableCode": false,
    "noFallthroughCasesInSwitch": true,
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noUnusedParameters": true,
    "noUnusedLocals": true,
    "skipLibCheck": true,
    "strictFunctionTypes": true,
    "strictNullChecks": false,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types",
      "@custom_types"
    ],
    "lib": [
      "es2017",
      "dom"
    ],
    "paths": {
      ...
      "@triu/ng-cms-core-api-service": [
        "dist/@triu/ng-cms-core-api-service"
      ]
      ...
    }
  }
}
```

The error is something like this:
```
ERROR in ./dist/@triu/ng-cms-core-api-service/fesm5/triu-ng-cms-core-api-service.js
Module not found: Error: Can't resolve '@triu/common/api/src/api-services' in '[project_directory]/dist/@triu/ng-cms-core-api-service/fesm5'
```

#### Version Information

```
$ node_modules/.bin/ng-packagr --version
ng-packagr:            3.0.0
@angular/compiler:     6.0.3
rollup:                0.59.4
tsickle:               0.29.0
typescript:            2.7.2
```

**Comment by alan-agius4** [comment]
Do you think you can provide an reproduction?

**Comment by maxtacco** [comment]
@alan-agius4 Unfortunately, the project I'm working on uses a bunch of private npm libraries and quite large. Once I have some time I'll try to create a repo, but I can't promise when this is going to happen. Meanwhile, we can see if anyone else hits the same issue. Worst case we can switch target in compileOptions to es2015 for now. Thank you for your support.

**Comment by alan-agius4** [comment]
Changing the target shouldn’t have any effect.  As the compilation target is set internally.

**Comment by alan-agius4** [comment]
Ah that version, I though the target version in the lib tsconfig.

**Comment by alan-agius4** [comment]
I tried to replicate this, with no luck

**Comment by maxtacco** [comment]
Let me add some source code extracts, may be it'll help understand the issue better.

This is the class that was used to generate the code from above:
```
...
import {Injectable} from '@angular/core';
import {ApiServices, ApiServiceSupport} from '@triu/common/api';
...

// @dynamic
@Injectable({
  providedIn: 'root'
})
export class CoreAccountService extends ApiServiceSupport {
  ...
  constructor(services: ApiServices,
              private localStorageService: LocalStorageService) {
    super(services, '/[some_endpoint_url]');
  }
  ...
```

Now, this class extends ApiServiceSupport class from  `@triu/common/api` that is installed as npm dependency. 

The `public_api.d.ts` in `node_modules/@triu/common/api` has something like this:
```
/**
 * @module
 * @description
 * Entry point for all public APIs of this package.
 */
...
export { ApiServices } from './src/api-services';
export * from './src/api-service-support';
...
```

`node_modules/@triu/common/api/src/api-services.d.ts` contains the following:
```
import { HttpClient } from '@angular/common/http';
import { Injector } from '@angular/core';
import { Logger } from '@triu/common/logger';
export declare class ApiServices {
    private injector;
    private _httpClient;
    private _logger;
    constructor(injector: Injector);
    getBaseUrl(): string;
    isApiUrl(url: string): boolean;
    isPublicApiUrl(url: string): boolean;
    readonly httpClient: HttpClient;
    readonly logger: Logger;
}
```
Finally, `node_modules/@triu/common/api/src/api-service-support.d.ts` for the parent class has the following:
```
import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
import { Logger } from '@triu/common/logger';
import { ApiServices } from './api-services';

export declare abstract class ApiServiceSupport {
    protected services: ApiServices;
    protected endpointPath: string;
    protected httpClient: HttpClient;
    protected logger: Logger;
    constructor(services: ApiServices, endpointPath: string);
    getApiPath(): string;
   ...
}
```
Here `ApiServices` are exported using a relative path: `import { ApiServices } from './api-services'`.  I think for some reason this is the cause of the duplicate import in the generated code from above. But, I'm not sure why it works with ng-packagr 3.0.0-rc.5 and not with 3.0.0

**Comment by alan-agius4** [comment]
Without a reproduction, this will be hard to track down.

And you mentioned that the duplicate import is happing both in FESM5 and ESM5 correct, not just the FESM file?

**Comment by maxtacco** [comment]
Yes, for ESM imports look a little bit different, but there is a duplicate still:
```
import { Injectable } from '@angular/core';
import { ApiServices, ApiServiceSupport } from '@triu/common/api';
...
import * as i0 from "@angular/core";
import * as i1 from "@triu/common/api/src/api-services";
...
```

**Comment by alan-agius4** [comment]
this tranformation is quite strange
```
import * as i0 from "@angular/core";
import * as i1 from "@triu/common/api/src/api-services";
```

As the output should be something like
```
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { ApiServices, ApiServiceSupport } from '@triu/common/api';
```

**Comment by maxtacco** [comment]
Also, I understand that there is a lot of things involved in the building process. So, we can wait until someone else hits the same issue and can reproduce it in a repo if necessary.

**Comment by maxtacco** [comment]
Let me add the parent tsconfig.lib.json we use for the libraries:
```
{
  "compilerOptions": {
    "declaration": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "allowUnreachableCode": false,
    "noFallthroughCasesInSwitch": true,
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noUnusedParameters": true,
    "noUnusedLocals": true,
    "skipLibCheck": true,
    "strictNullChecks": true,
    "stripInternal": true,
    "sourceMap": true,
    "inlineSources": true,
    "importHelpers": true,
    "pretty": true,
    "moduleResolution": "node",
    "module": "es2015",
    "target": "es2015",
    // don't auto-discover @types/node
    "types": [],
    "lib": [
      "es2017",
      "dom"
    ]
  },
  "angularCompilerOptions": {
    "annotateForClosureCompiler": true,
    "fullTemplateTypeCheck": true,
    "skipTemplateCodegen": true,
    "strictMetadataEmit": true,
    "strictInjectionParameters": true,
    "flatModuleId": "AUTOGENERATED",
    "flatModuleOutFile": "AUTOGENERATED"
  },
  "exclude": [
    "src/test.ts",
    "**/*.spec.ts"
  ]
}
```

And this is the one for the library itself:
```
{
  "extends": "../../../tsconfig.lib.json",
  "compilerOptions": {
    "baseUrl": "",
    "paths": {
      ...
    }
  }
}

```

**Comment by Judgeja** [comment]
Not sure if it's the same issue, but I was having the same error message with my Angular 6 private repo. 
For me it was because I had cd'd into /projects/mylib and done npm installing for the lib in the package.json there instead of the project root one, which created a separate node_modules folder that isn't found by the files in the /dist folder.

**Comment by maxtacco** [comment]
@Judgeja, I think it is not the same case here. I'm building libraries from the root project of the folder and I don't use npm install/link - instead I have dependencies specified in the "paths" option of app's tsconfig.json.

**Comment by maxtacco** [comment]
@alan-agius4 Ok, I finally was able to figure out the issue with our setup. We have a custom `tsconfig.lib.json` in the root folder of the project that all our library `tsconfig.lib.json` files extend from. But, according to this issue https://github.com/angular/angular-cli/issues/8336 "angularCompilerOptions" section of the config is not extended. So, we had to copy "angularCompilerOptions" to all of our library  tsconfig.lib.json files and that fixed ESM5/FESM5 module generation. 
Thank you all for the help.
